analyzer/runtime/abiparse/abiparse.go

```diff
@@ -87,7 +87,7 @@ func ParseEvent(topics [][]byte, data []byte, contractABI *abi.ABI) (*abi.Event,
 	}
 	event, err := contractABI.EventByID(topicsEC[0])
 	if err != nil {
-		return nil, nil, fmt.Errorf("contract ABI EventByID: %w", err)
+		return nil, nil, fmt.Errorf("contract ABI EventByID: %w, eventID: %s", err, topicsEC[0].Hex())
```

Comment:
the [go-eth EventByID](https://github.com/ethereum/go-ethereum/blob/v1.13.14/accounts/abi/abi.go#L223) mangles the event signature so we explicitly log it here

Reply:
> mangles

> 	return nil, fmt.Errorf("no event with id: %#x", topic.Hex())

what does it do?

Reply:
`topic.Hex()` returns a hex `string` which is then interpreted as utf-8 bytes and converted again to hex bytes (`%#x`). tldr the topic becomes this ~140 char long hex string.

Reply:
their bug huh, rip